packages/fast-components-react-base/src/checkbox/checkbox.tsx

```diff
@@ -87,6 +88,34 @@ class Checkbox extends Foundation<ICheckboxHandledProps & ICheckboxManagedClasse
 
         return super.generateClassNames(classes);
     }
+
+    /**
+     * Stores HTML tag for use in render
+     */
+    private get tag(): string {
+        return CheckboxHTMLTags[this.props.tag] || CheckboxHTMLTags.label;
+    }
+
+    /**
+     * Apply indeterminate state to items that are indeterminate.
+     * This method should be called after render because it relies on element references.
+     */
+    private applyIndeterminateState(): void {
+         if (!isUndefined(this.props.indeterminate) && this.inputRef.current && this.state.checked) {
+            this.inputRef.current.indeterminate = this.props.indeterminate;
+        }
+    }
+
+    /**
+     * Handles onChange as a controlled component
+     */
+    private handleCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        this.setState({checked: !this.state.checked});
```

Comment:
The only issue now is if an author sets the `checked` property, and the component is clicked by the user, the property is then inconsistent with the state of the UI.

Props should be the single source of truth - if an author sets the checked prop, the UI needs to represent that value until the author changes it. It should not update internally if it is being controlled from the outside.

Reply:
+1, this is the difference between controlled and uncontrolled